src/storage/hummock_sdk/src/table_watermark.rs

```diff
@@ -507,36 +535,32 @@ impl TableWatermarks {
         }
         debug!("clear stale table watermark below epoch {}", safe_epoch);
         let mut result_epoch_watermark = Vec::with_capacity(self.watermarks.len());
-        let mut unset_vnode: HashSet<VirtualNode> = (0..VirtualNode::COUNT)
-            .map(VirtualNode::from_index)
-            .collect();
+        let mut set_vnode: HashSet<VirtualNode> = HashSet::new();
         while let Some((epoch, _)) = self.watermarks.last() {
             if *epoch >= safe_epoch {
                 let (epoch, watermarks) = self.watermarks.pop().expect("have check Some");
                 for watermark in watermarks.as_ref() {
                     for vnode in watermark.vnode_bitmap.iter_vnodes() {
-                        unset_vnode.remove(&vnode);
+                        set_vnode.insert(vnode);
                     }
                 }
                 result_epoch_watermark.push((epoch, watermarks));
             } else {
                 break;
             }
         }
-        while !unset_vnode.is_empty()
```

Comment:
Remove this check seems incorrect. @wenym1 PTAL

Reply:
Re-read the logic and the implementation in this PR is correct because all watermarks share the same vnode count.